mlflow/entities/dataset_input.py

```diff
@@ -0,0 +1,43 @@
+from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.protos.service_pb2 import DatasetInput as ProtoDatasetInput
+from mlflow.entities.dataset import Dataset
+from mlflow.entities.input_tag import InputTag
+
+
+class DatasetInput(_MLflowObject):
+    """DatasetInput object associated with an experiment."""
+
+    def __init__(self, dataset, tags=[]):
+        self._dataset = dataset
+        self._tags = tags
```

Comment:
Nit (Python purists discourage using mutable types like lists as function or constructor parameters):

```suggestion
    def __init__(self, dataset, tags=None):
        self._dataset = dataset
        self._tags = tags or []
```